src/develop/masks.h

```diff
@@ -223,6 +223,7 @@ typedef struct dt_masks_form_gui_t
 
 
   gboolean creation;
+  gboolean creation_pause;
```

Comment:
This boolean state help keep track when left mouse button released, but CTRL key has been pressed, to continue mask creation mode, but don't add any additional points when mouse pointer moves. After the secondary left button pressed - mask points are started to adding. After releasing left button and before pressing - there are two additional points added - with zero line (mask) size.

Reply:
I see, not fully obvious from code reading, maybe adding some comments will help.

Reply:
If there are some other state that we can use to track this kind of event, without breaking current experience - it will be the best to use such state, but I can't find anything for similar action.